ros2param/test/test_cli.py

```diff
@@ -0,0 +1,718 @@
+            strict=True
+        )
+
+    # Get verb tests without --hide-type flag
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_invalid_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['get', '/param_node', 'unexistent_parameter']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=[
+                'Parameter not set'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_valid_double_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['get', '/param_node', 'double_param']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Double value is: 1.23'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_valid_boolean_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['get', '/param_node', 'bool_param']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Boolean value is: True'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_valid_int_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['get', '/param_node', 'int_param']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['Integer value is: 42'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_valid_string_param_existing_node(self):
+        with self.launch_node_command(
+                arguments=['get', '/param_node', 'str_param']) as node_command:
+            assert node_command.wait_for_shutdown(timeout=5)
+        assert node_command.exit_code == launch_testing.asserts.EXIT_OK
+        assert launch_testing.tools.expect_output(
+            expected_lines=['String value is: Hello World'],
+            text=node_command.output,
+            strict=True
+        )
+
+    @launch_testing.markers.retry_on_failure(times=3)
+    def test_get_valid_string_array_param_existing_node(self):
```

Comment:
@BMarchi meta: I wonder if we can parameterize these `ros2 param get` tests somehow, it's pretty much the same code except for `arguments` and `expected_lines`.